sql/2024/fonts/common.sql

```diff
@@ -0,0 +1,94 @@
+-- Normalize a name. Used in FAMILY.
+CREATE TEMPORARY FUNCTION FAMILY_INNER(name STRING) AS (
+  IF(LENGTH(TRIM(name)) < 3, NULL, NULLIF(TRIM(name), ''))
+);
+
+-- Extract the family name from a payload.
+CREATE TEMPORARY FUNCTION FAMILY(payload STRING) AS (
+  FAMILY_INNER(JSON_EXTRACT_SCALAR(payload, '$._font_details.names[1]'))
+);
+
+-- Extract the file format from a URL and a Content-Type header.
+CREATE TEMPORARY FUNCTION FILE_FORMAT(url STRING, header STRING) AS (
+  LOWER(COALESCE(
```

Comment:
This is different from previous years, what's the rationale for the change? Are they functionally equivalent? If not, we should make a note in the chapter explaining why it has been changed.

Reply:
The percentages are quite different from 2022, so we should definitely look into it.

Reply:
You mean this, right?

https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2022/fonts/font_format_usage.sql#L5

There is no `mimeType` in `requests` anymore, but there are request headers. So I take `Content-Type`:

https://github.com/IvanUkhov/almanac.httparchive.org/blob/iu-fonts/sql/2024/fonts/performance/fonts_format_file.sql#L18-L19

Reply:
And there is no more `ext` either. So I take the show URL.

https://github.com/IvanUkhov/almanac.httparchive.org/blob/iu-fonts/sql/2024/fonts/performance/fonts_format_file.sql#L8

Reply:
Added two pages showing the same two queries but with the original regular expression. Please take a look:

<img width="612" alt="Skärmavbild 2024-08-12 kl  09 43 00" src="https://github.com/user-attachments/assets/f2b4b597-4553-4509-b32b-b280ddd7ace5">

Reply:
@IvanUkhov 
`mimeType` and `ext` still exists, it moved to `summary` in requests: https://har.fyi/reference/structs/request-summary/
So yo can do `JSON_VALUE(summary, '$.mimeType') as mimeType` to access the mimeType and any other value of `summary`.

Reply:
Thank you, Jannis! Totally missed that. I have now updated both the two queries and the two pages in the spreadsheet. I will also update the queries regarding the file size; I also took that from the headers. Instead of the summary field.